### PR TITLE
Likes & watch later

### PR DESCRIPTION
**Pull request by Digambar1411**
add likes and watch later functionality

### CODE REVIEW

File: src/frontend/components/smallVideoCard/smallVideoCard.css [hunk 1]
```diff
@@ -1,5 +1,6 @@
 .video-card{
     height:245px;
```
Comment (aurobindoGupta) [inline src/frontend/components/smallVideoCard/smallVideoCard.css:2]: rem

**Review by aurobindoGupta — APPROVED** [review]
good work, keep it up.

File: src/App.js [hunk 2]
```diff
@@ -9,11 +10,36 @@ function App() {
...
+        />
+        <Route path="/play-list" 
+          element={
+            <PrivateRoutes >
+              <Playlist />
+            </PrivateRoutes>
+          }
+        />
+        
         <Route path="/user-profile" element={<UserProfle />} />
```
Comment (sashaboi) [inline src/App.js:43]: do make sure to add 404 route before publishing

File: src/frontend/components/bigHorizontalVideoCard/bigHorizontalVideoCard.js [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+    const isInWatchLater = watchLaterVideos.find(watchLaterVideo=>watchLaterVideo._id===video._id ? true : false)
+    const isInLikedVideos = likedVideos.find(likedVideo=>likedVideo._id===video._id ? true : false)
+
+    
+    const addToWatchLaterVideos = async()=>{
+        if(isLoggedIn){
+            try{
+                const response = await addToWatch(video,token)
+                console.log(response)
```
Comment (sashaboi) [inline src/frontend/components/bigHorizontalVideoCard/bigHorizontalVideoCard.js:28]: clean up consoles before releasing

File: src/frontend/components/bigHorizontalVideoCard/bigHorizontalVideoCard.js [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+    const addToLikedVideos = async()=>{
+
+        try{
+            const response = await addToLikeService(video, token)
+            if(response.status===201){
+                setLikedVideos(response.data.likes)
+            }
+        }
+        catch(error){
+            console.error(error)
```
Comment (sashaboi) [inline src/frontend/components/bigHorizontalVideoCard/bigHorizontalVideoCard.js:52]: try using some toasts library . this shows foresight for interviewer. 
if backed breaks, frontend isnt blank , it shows error thats good UX

File: src/frontend/components/bigHorizontalVideoCard/bigHorizontalVideoCard.js [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+                    <div className="sm-card-title">{video.title}</div>
+                    <div className='sm-card-creator light-font'>{video.creator}</div>
+                </div>
+                
+                <span className="material-icons-outlined" onClick={toggleModal}>
+                    more_vert
+                </span>
+
+        
+        { showModal &&
```
Comment (sashaboi) [inline src/frontend/components/bigHorizontalVideoCard/bigHorizontalVideoCard.js:99]: i can have two modals open at the same time , doesnt make sense. 
image
do fix before publishing

File: src/frontend/components/bigHorizontalVideoCard/bigHorizontalVideoCard.js [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+                    <div className='sm-card-creator light-font'>{video.creator}</div>
+                </div>
+                
+                <span className="material-icons-outlined" onClick={toggleModal}>
+                    more_vert
+                </span>
+
+        
+        { showModal && 
+            <div className="modal">
```
Comment (sashaboi) [inline src/frontend/components/bigHorizontalVideoCard/bigHorizontalVideoCard.js:100]: image

z indexes for modal and top navbar seem to clash do take care before publishing

File: src/frontend/components/index.js [hunk 7]
```diff
@@ -4,5 +4,8 @@ import { Sidebar} from "./sidebar/sidebar";
 import { SmallVideoCard } from "./smallVideoCard/smallVideoCard";
 import { CategoryBar } from "./categoryBar/categoryBar";
 import { SmallHorizontalVideoCard } from "./smallHorizontalVideoCard/smallHorizontalVideoCard"
+import { BigHorizontalCard} from "./bigHorizontalVideoCard/bigHorizontalVideoCard"
```
Comment (sashaboi) [inline src/frontend/components/index.js:7]: image


aspect ratio of thumbnails seem to be off , do take care , in liked videos page

File: src/frontend/components/smallVideoCard/smallVideoCard.js [hunk 8]
```diff
@@ -1,19 +1,60 @@
 import Moment from 'react-moment';
```
Comment (sashaboi) [inline src/frontend/components/smallVideoCard/smallVideoCard.js:1]: Here, I am unable to click on text title of video to go to single video page. maybe the link / navigate could encompass the entire card than just the thumbnail ? just nitpicking ux

**Review by sashaboi — APPROVED** [review]
I love the styling efforts , color scheme , good use of icons, and functionality you have been able to achieve. 
While going for job in roc8 , and while doing job, I notice ux and small small things a lot more, which might feel unnecessary for you at this point.

I have made some non mandatory suggestions. you can try doing those when you find time after submission. 

This version looks good to merge to me , great work !

suggestion : 

image

try designing for screens of all sizes, mine is a full hd screen, looks a little broken :) 
check responsive 1920*1080 , 1260*720 , and other common devices. top startups use 4k screens so keep those dimensions too.

**Comment by Digambar1411** [comment]
> I love the styling efforts , color scheme , good use of icons, and functionality you have been able to achieve. While going for job in roc8 , and while doing job, I notice ux and small small things a lot more, which might feel unnecessary for you at this point.
> 
> I have made some non mandatory suggestions. you can try doing those when you find time after submission.
> 
> This version looks good to merge to me , great work !
> 
> suggestion :
> 
> image
> 
> try designing for screens of all sizes, mine is a full hd screen, looks a little broken :) check responsive 1920_1080 , 1260_720 , and other common devices. top startups use 4k screens so keep those dimensions too.

Really thanks for detail review in the next pr I m going to do responsive that time I will fix and implement the suggestions

**Pull request merged**